lib/util.js

```diff
@@ -430,3 +432,149 @@ module.exports = exports = {
+              if (config.events.includes('signal'))
+                process.on(options.signal, handleSignal);
+              config.signal = options.signal;
+              break;
+            case '3':
+              if (typeof options.filename !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('filename',
+                                               'String',
+                                               options.filename);
+              config.filename = options.filename;
+              break;
+            case '4':
+              if (typeof options.path !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('path', 'String', options.path);
+              config.path = options.path;
+              break;
+            case '5':
+              if (typeof options.verbose !== 'string' &&
+                  typeof options.verbose !== 'boolean')
+                throw new ERR_INVALID_ARG_TYPE('verbose',
+                                               'Booelan | String' +
+                                               ' (true|false|yes|no)',
+                                               options.verbose);
+              config.verbose = options.verbose;
+              break;
+          }
+        });
+        // upload this new config to C++ land
+        nr.syncConfig(config, true);
+      },
+
+
+      triggerReport: function(file, err) {
+        if (err === undefined || err == null) {
+          if (file === undefined || file == null)
+            return nr.triggerReport(new ERR_SYNTHETIC(
+              'JavaScript Callstack').stack);
+          if (typeof file !== 'string')
+            throw new ERR_INVALID_ARG_TYPE('file', 'String', file);
+          return nr.triggerReport(file, new ERR_SYNTHETIC(
+            'JavaScript Callstack').stack);
+        }
+        if (typeof err !== 'object')
+          throw new ERR_INVALID_ARG_TYPE('err', 'Object', err);
+        if (file === undefined || file == null)
+          return nr.triggerReport(err.stack);
+        if (typeof file !== 'string')
+          throw new ERR_INVALID_ARG_TYPE('file', 'String', file);
+        return nr.triggerReport(file, err.stack);
+      },
+      getReport: function(err) {
+        if (err === undefined || err == null) {
+          return nr.getReport(new ERR_SYNTHETIC('JavaScript Callstack').stack);
+        } else if (typeof err !== 'object') {
+          throw new ERR_INVALID_ARG_TYPE('err', 'Objct', err);
+        } else {
+          return nr.getReport(err.stack);
+        }
+      }
+    };
+
+    // download the CLI / ENV config into JS land.
+    nr.syncConfig(config, false);
+
+    function handleSignal(signo) {
+      if (typeof signo !== 'string')
+        signo = config.signal;
+      nr.onUserSignal(signo);
+    }
+
+    if (config.events.includes('signal')) {
+      process.on(config.signal, handleSignal);
+    }
+
+    function parseOptions(obj) {
+      const list = [];
+      if (obj == null || obj === undefined)
```

Comment:
`obj == null` will catch the `undefined` cases also

Reply:
done, used nullchecks across the function, and wherever applicable.